ext/soap/php_encoding.c

```diff
@@ -1259,6 +1259,7 @@ static void model_to_zval_any(zval *ret, xmlNodePtr node)
 					ZVAL_NULL(&val2);
 					master_to_zval(&val2, get_conversion(XSD_ANYXML), node->next);
 					if (Z_TYPE(val2) != IS_STRING ||  *Z_STRVAL(val) != '<') {
+						Z_TRY_DELREF(val2);
```

Comment:
When I change it to `zval_ptr_dtor(&val2)`, I consistently get a segfault. It looks like calling the destructor invalidates an object that was stored in `SOAP_GLOBAL(ref_map)`, and that object is needed (by `soap_check_xml_ref`) on the next iteration of the outer loop.